agent/engine/execcmd/manager_start_linux.go

```diff
@@ -135,35 +140,45 @@ func (m *manager) doStartAgent(ctx context.Context, client dockerapi.DockerClien
 	}
 	execRes, err := client.CreateContainerExec(ctx, containerId, execCfg, dockerclient.ContainerExecCreateTimeout)
 	if err != nil {
+		execMD.StoppedReason = fmt.Sprintf("create start error: %v", err)
```

Comment:
hmm, in these case the StoppedReason will be lost because `SetExecCommandAgentMetadata` is never invoked. I'm thinking invoking `SetExecCommandAgentMetadata` and  `SetKnownExecCommandAgentStatus` should be invoked in `m.StartAgent` rather than here.